File: backend/core/rabbitmq.js
const amqp = require('amqplib');

let channel = null;

const connectRabbitMQ = async () => {
  try {
    const amqpServer = `amqp://${process.env.RABBITMQ_USER}:${process.env.RABBITMQ_PASS}@${process.env.RABBITMQ_HOST}`;
    const connection = await amqp.connect(amqpServer);
    channel = await connection.createChannel();
    await channel.assertQueue('orders_queue', { durable: true });
    console.log('RabbitMQ connected');
  } catch (error) {
    console.error('RabbitMQ connection failed:', error);
  }
};

connectRabbitMQ();

const publish = (queue, msg) => {
  if (!channel) {
    console.error('RabbitMQ channel not established');
    return;
  }
  channel.sendToQueue(queue, Buffer.from(JSON.stringify(msg)), { persistent: true });
  console.log(`Message sent to ${queue}`);
};

module.exports = { publish };

File: backend/core/rabbitmq.py
import pika
import json
import logging
from core.config import settings

logger = logging.getLogger(__name__)

class RabbitMQClient:
    def __init__(self):
        self.connection = None
        self.channel = None

    def connect(self):
        if not self.connection or self.connection.is_closed:
            credentials = pika.PlainCredentials(settings.RABBITMQ_USER, settings.RABBITMQ_PASS)
            parameters = pika.ConnectionParameters(
                host=settings.RABBITMQ_HOST,
                port=settings.RABBITMQ_PORT,
                credentials=credentials,
                heartbeat=600
            )
            try:
                self.connection = pika.BlockingConnection(parameters)
                self.channel = self.connection.channel()
                # Declare essential queues
                self.channel.queue_declare(queue='orders_queue', durable=True)
                self.channel.queue_declare(queue='training_queue', durable=True)
                logger.info("Connected to RabbitMQ")
            except Exception as e:
                logger.error(f"Failed to connect to RabbitMQ: {e}")

    def publish(self, queue_name: str, message: dict):
        self.connect()
        try:
            self.channel.basic_publish(
                exchange='',
                routing_key=queue_name,
                body=json.dumps(message),
                properties=pika.BasicProperties(
                    delivery_mode=2,  # Make message persistent
                )
            )
            logger.info(f"Message published to {queue_name}")
        except Exception as e:
            logger.error(f"Failed to publish message: {e}")

    def close(self):
        if self.connection and not self.connection.is_closed:
            self.connection.close()

mq_client = RabbitMQClient()